### ISSUE TITLE
MyTopLevelSim Failure on Windows

### ISSUE THREAD

**Issue by krauslabs** [SpinalHDL/SpinalTemplateSbt#2]
Trying to run the template project on Windows 10, and I am able to successfully compile the Spinal code into VHDL or Verilog, but running the simulation testbench causes failures.

Here is a snippet from the output describing the failure:

```
[info] [Progress] at 0.190 :   emit MyTopLevel
[info] [Info] Number of registers : 8
[info] [Done] at 0.246
[info] [Progress] Simulation workspace in D:\projects\spinal\SpinalTemplateSbt\.\simWorkspace\MyTopLevel
[info] [Progress] Verilator compilation started
[error] Exception in thread "main" java.io.FileNotFoundException: Source 'C:\Program Files\Java1.8.0_181\include' does not exist
[error]         at org.apache.commons.io.FileUtils.copyDirectory(FileUtils.java:1368)
[error]         at org.apache.commons.io.FileUtils.copyDirectory(FileUtils.java:1261)
[error]         at org.apache.commons.io.FileUtils.copyDirectory(FileUtils.java:1230)
[error]         at spinal.sim.VerilatorBackend.compileVerilator(VerilatorBackend.scala:322)
[error]         at spinal.sim.VerilatorBackend.<init>(VerilatorBackend.scala:405)
[error]         at spinal.core.sim.SpinalVerilatorBackend$.apply(SimBootstraps.scala:117)
[error]         at spinal.core.sim.SpinalSimConfig.compile(SimBootstraps.scala:374)
[error]         at spinal.core.sim.SpinalSimConfig.compile(SimBootstraps.scala:338)
[error]         at spinal.core.sim.SpinalSimConfig.doSim(SimBootstraps.scala:325)
[error]         at mylib.MyTopLevelSim$.main(MyTopLevelSim.scala:13)
[error]         at mylib.MyTopLevelSim.main(MyTopLevelSim.scala)
java.lang.RuntimeException: Nonzero exit code returned from runner: 1
        at scala.sys.package$.error(package.scala:27)
[trace] Stack trace suppressed: run last compile:runMain for the full output.
[error] (compile:runMain) Nonzero exit code returned from runner: 1
[error] Total time: 9 s, completed Sep 12, 2018 10:45:42 PM
```

It seems to successfully generate the verilog code from the module, but then fails to compile it through verilator, but in a way related to Java (`Source 'C:\Program Files\Java1.8.0_181\include' does not exist`), which seems odd.

Some information about the system is included below:

- Windows 10
- JDK and JRE 8u181 both installed (in directory `C:\Program Files\Java\jdk1.80_181`, so the error up above seems to have an incorrect path)
- SBT installed, version 1.2.1 installed (it uses 0.13.16 for the template)
- Verilator installed through prebuilt MinGW binaries (version 3.841)

Let me know if there is any additional info that would help diagnose what may be going on here.

**Comment by Dolu1990** [comment]
Hi,
I made a w10 setup few days ago, i had the same error message, and basicaly, what i had to change in the environnement, is the PATH variable.
Be sure that the  JDK/bin is pointed by the PATH variable, and not the JRE/xx
It should solve the issue, tell me if not.
For verilator, i think with the version that you are using you will get some errors. I updated the documentation just now to get a specific version of Verilator. See : https://github.com/SpinalHDL/SpinalDoc/blob/gh-pages/spinal/sim/introduction.md#windows

**Comment by Dolu1990** [comment]
What happen when you do a "where java"   and a "where javac" ?

**Comment by krauslabs** [comment]
Yep the points you made definitely seem to be causing the issue. Changing the path to point to the JDK/bin folder (before `oracle/classpath/java` which was its default) fixed the Java issues.

I then did run into verilator issues like you predicted, so thank you for the heads up. I found out that the verilator version that can be installed through MSYS2 pacman will also work correctly, so installation is as easy as `pacman -S mingw-w64-x86_64-verilator` and adding the msys2/usr/bin and mingw64/bin to the windows path.

Thanks for all your help!
